data/data-pipeline/data_pipeline/comparison_tool/src/utils.py

```diff
@@ -0,0 +1,330 @@
+import pandas as pd
+from data_pipeline.score import field_names
+
+
+FIPS_MAP = {
+    "53": "WA",
+    "10": "DE",
+    "11": "DC",
+    "55": "WI",
+    "54": "WV",
+    "15": "HI",
+    "12": "FL",
+    "56": "WY",
+    "72": "PR",
+    "34": "NJ",
+    "35": "NM",
+    "48": "TX",
+    "22": "LA",
+    "37": "NC",
+    "38": "ND",
+    "31": "NE",
+    "47": "TN",
+    "36": "NY",
+    "42": "PA",
+    "02": "AK",
+    "32": "NV",
+    "33": "NH",
+    "51": "VA",
+    "08": "CO",
+    "06": "CA",
+    "01": "AL",
+    "05": "AR",
+    "50": "VT",
+    "17": "IL",
+    "13": "GA",
+    "18": "IN",
+    "19": "IA",
+    "25": "MA",
+    "04": "AZ",
+    "16": "ID",
+    "09": "CT",
+    "23": "ME",
+    "24": "MD",
+    "40": "OK",
+    "39": "OH",
+    "49": "UT",
+    "29": "MO",
+    "27": "MN",
+    "26": "MI",
+    "44": "RI",
+    "20": "KS",
+    "30": "MT",
+    "28": "MS",
+    "45": "SC",
+    "21": "KY",
+    "41": "OR",
+    "46": "SD",
+}
+
+
+def read_file(
+    file_path: str, columns: list, geoid: str = field_names.GEOID_TRACT_FIELD
+) -> pd.DataFrame:
+    """Reads standardized csvs in
+
+    Parameters:
+        file_path: the file to read
+        columns: the columns to include
+        geoid: the geoid column name (if we change this in usa.csv, will need to change this slightly)
+
+    Returns:
+        dataframe that has been read in with geographic index
+    """
+    assert (
+        geoid == field_names.GEOID_TRACT_FIELD
+    ), f"Field name specified for geoid is incorrect. Use {field_names.GEOID_TRACT_FIELD}"
+    return pd.read_csv(
+        file_path, usecols=columns + [geoid], dtype={geoid: str}
+    ).set_index(geoid)
+
+
+def produce_summary_stats(
+    joined_frame: pd.DataFrame,
+    comparator_column: str,
+    score_column: str,
+    population_column: str,
+    geoid_column: str = field_names.GEOID_TRACT_FIELD,
+) -> pd.DataFrame:
+    """Produces high-level overview dataframe
+
+    Parameters:
+        joined_frame: the big df
+        comparator_column: the column name for the comparator identification bool
+        score_column: the column name for the CEJST score bool
+        population_column: the column that includes population count per tract
+        geoid_column: the geoid10_tract column
+
+    Returns:
+        population: the high-level overview df
+    """
+    population = joined_frame.groupby([comparator_column, score_column]).agg(
```

Comment:
Nit: I have a (unwritten!) style preference of ending all dataframe names with `_df`. I see you sometimes do this with `_frame`. Should we align on a single style?

Reply:
`_df` is totally fine by me!